main.c

```diff
@@ -666,11 +689,111 @@ static int display_event(int fd, uint32_t mask, void *data) {
 	return count;
 }
 
+static char *get_config_path(void) {
+	static char *config_paths[3] = {
+		"$XDG_CONFIG_HOME/swayidle/config",
+		"$HOME/.swayidle/config",
+		SYSCONFDIR "/swayidle/config",
+	};
+
+	char *config_home = getenv("XDG_CONFIG_HOME");
+
+	if (!config_home || config_home[0] == '\n') {
+		config_paths[0] = "$HOME/.config/swayidle/config";
+	}
+
+	wordexp_t p;
+	char *path;
+	for (size_t i = 0; i < sizeof(config_paths) / sizeof(char *); ++i) {
+		if (wordexp(config_paths[i], &p, 0) == 0) {
+			path = strdup(p.we_wordv[0]);
+			wordfree(&p);
+			if (path && access(path, R_OK) == 0) {
+				return path;
+			}
+			free(path);
+		}
+	}
+
+	return NULL;
+}
+
+static int load_config(const char *config_path) {
+	FILE *f = fopen(config_path, "r");
+
+	if (!f) {
+		swayidle_log(LOG_ERROR, "Failed to read config. Starting without it...");
+		return -1;
+	}
+
+	size_t lineno = 0;
+	char *line = NULL;
+	size_t n = 0;
+	ssize_t nread;
+	while ((nread = getline(&line, &n, f)) != -1) {
+		lineno++;
+		line[nread-1] = '\0';
+
+		if (strlen(line) == 0 || line[0] == '#') {
+			continue;
+		}
+
+		size_t i = 0;
+		while (line[i] != '\0' && line[i] != ' ') {
+			i++;
+		}
+
+		wordexp_t p;
+		wordexp(line, &p, 0);
+		if (strncmp("timeout", line, i) == 0) {
+			parse_timeout(p.we_wordc, p.we_wordv);
+		} else if (strncmp("before-sleep", line, i) == 0) {
+			parse_sleep(p.we_wordc, p.we_wordv);
+		} else if (strncmp("after-resume", line, i) == 0) {
+			parse_resume(p.we_wordc, p.we_wordv);
+		} else if (strncmp("lock", line, i) == 0) {
+			parse_lock(p.we_wordc, p.we_wordv);
+		} else if (strncmp("unlock", line, i) == 0) {
+			parse_unlock(p.we_wordc, p.we_wordv);
+		} else if (strncmp("idlehint", line, i) == 0) {
+			parse_idlehint(p.we_wordc, p.we_wordv);
+		} else {
+			line[i] = 0;
+			swayidle_log(LOG_ERROR, "Unexpected keyword \"%s\" in line %lu", line, lineno);
+			free(line);
+			return -1;
+		}
+		wordfree(&p);
+	}
+	free(line);
+	fclose(f);
+
+	return 0;
+}
+
+
 int main(int argc, char *argv[]) {
-	if (parse_args(argc, argv) != 0) {
+	swayidle_init();
+	char *config_path = NULL;
+	if (parse_args(argc, argv, &config_path) != 0) {
+		swayidle_finish();
 		return -1;
 	}
 
+	if (!config_path) {
+		config_path = get_config_path();
+	}
+
+	if (load_config(config_path) == -1) {
+		swayidle_finish();
+		swayidle_init();
+		swayidle_log(LOG_ERROR, "Failed to load config. Starting without it...");
```

Comment:
`load_config` also logs this error, causing a duplicate log entry.

Also this should probably use `LOG_DEBUG` instead of `LOG_ERROR`, considering it's intentional that users be able to launch it without a config file.

Reply:
Changed to `LOG_DEBUG` and logging now takes place in the callee.